### ISSUE TITLE
修改后模型不收敛

### ISSUE THREAD

**Issue by mkbk-with-circle** [netiken/m3#8]
Dear Aurthor

I hope this message finds you well. I have been working with your model and have made some modifications to the __getitem__ function in the PathDataset_Context class within the dataset.py file. Specifically, instead of using the sldn to generate the res feature map, I am directly using the flow's fct to generate the feature map, which is then input to the model for training and learning.

My objective is to adapt the model so that it can generate the true fct values, as ultimately, I need to predict the maximum fct. However, after making these changes and following your "Train your own model" instructions to train the model, I am encountering an issue where the model does not seem to converge, at least within the first 100 epochs. Specifically, the following values remain constant:

val_loss_sync_step = 1.240
val_loss_sync_epoch = 1.150
train_loss_sync_epoch = 1.200
Meanwhile, the train_loss_sync_step fluctuates around 1.0.

I was wondering if this issue might be related to the dataset size, as shown in the attached image. Do you think the limited amount of data might be causing this behavior, or could it be due to another factor?

<img width="771" alt="Image" src="https://github.com/user-attachments/assets/adc2edd5-901c-4d05-aa34-a1110b7c0e43" />

I would greatly appreciate any guidance or suggestions you may have to help resolve this issue.

Thank you very much for your time and assistance.

Best regards,
mkbk_circle

**Comment by mkbk-with-circle** [comment]
this is my changed dataset.py (sorry to use .txt format because .py is not allowed to upload)

[dataset.py.txt](https://github.com/user-attachments/files/18724346/dataset.py.txt)

**Comment by mkbk-with-circle** [comment]
Oh, one supplement： Now I haven't add the normalization of the fct data,but I don't think it's the main reason for non-convergence.(of course I could be wrong but I still wish to get some advice from you ,thanks!)

**Comment by mkbk-with-circle** [comment]
或者对于我想得到最大值fct的情况，您有什么更好的建议吗？

**Comment by mkbk-with-circle** [comment]
fine,I add the normalization process but non-convergence still exist.......(You can see in my dataset.py.txt)
My normalization process is : read all the fct_flowsim data in each shard directory below 'dir_input' and calculate the mean and overall_variance.And each time I call the method __getitem__ , I will use these two data to normalize the data.

**Comment by liecn** [comment]
Please check the range of the ground truth data (i.e., FCT from ns-3). I have a feeling that the range might be too large.

Maybe try applying a logarithmic transformation to the ground truth data.

I don’t have a strong intuition about the data size, but at this stage, I don’t think it’s the main issue.

**Comment by mkbk-with-circle** [comment]
ok, I got it , I will try your method, thanks for your answer!

**Comment by mkbk-with-circle** [comment]
OK, now I know that it's because I use the wrong method of normalization.Thanks!
